### DISCUSSION TITLE
the problem of front-end(html/js) and back-end(java) interaction

### DISCUSSION THREAD

**Discussion by hanxiaohuivc** [tauri-apps/tauri#2404] [Q&A] [answered]
Hello everyone. I have a project that includes front end(The front end is a web page written by html, css, js) and back endback end is a business system written in java）, and the front and back ends use ‘ajax’ for interaction. If I use tauri to package the front-end files, does the interaction between the app.exe and the back-end need to be rewritten using tauriAPI?
When I use the "yarn tauri dev" command to debug, I can log in app.exe to view database information and other operations. However, when I run the exe generated by "yarn tauri build", I cannot log in.

**Comment by nothingismagick** [comment]
Please edit your question and explain the actual architecture of your "back ends".

**Comment by hanxiaohuivc** [comment, marked as answer]
I successfully tested tauri to interact with rust, and js to access webapi.

**Reply by ppazos** [reply]
can you explain what you did with webapi?
